containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -1,116 +1,326 @@
 import json
 import traceback
+from copy import deepcopy
+from typing import Dict
 
 import boto3
 import botocore
 from data_platform_logging import DataPlatformLogger
-from data_platform_paths import DataProductConfig
+from data_platform_paths import (
+    BucketPath,
+    DataProductConfig,
+    JsonSchemaName,
+    specification_path,
+    specification_prefix,
+)
 from dataengineeringutils3.s3 import get_filepaths_from_s3_folder, read_json_from_s3
 from jsonschema import validate
 from jsonschema.exceptions import ValidationError
 
 s3_client = boto3.client("s3")
 
+glue_csv_table_input_template = {
+    "DatabaseName": "",  # add to this from data product name pulled from api path
+    "TableInput": {
+        "Name": "",  # add to this from table name pulled from api path
+        "Description": "",
+        "Owner": "owner",
+        # "Retention": 0,
+        "StorageDescriptor": {
+            "Columns": [],  # add to this from schema passed by user in api request body
+            "Location": "",  # noqa E501 add to this once data land and timestamp generated, although could be top level bucket/data/product/version/table
+            "InputFormat": "org.apache.hadoop.mapred.TextInputFormat",
+            "OutputFormat": "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat",
+            "Compressed": False,
+            "NumberOfBuckets": -1,
+            "SerdeInfo": {
+                "SerializationLibrary": "org.apache.hadoop.hive.serde2.OpenCSVSerde",
+                "Parameters": {"field.delim": ",", "escape.delim": "\\"},
+            },
+            "BucketColumns": [],
+            "SortColumns": [],
+            "Parameters": {},
+            "StoredAsSubDirectories": False,
+        },
+        "PartitionKeys": [],
+        "TableType": "EXTERNAL_TABLE",
+        "Parameters": {"classification": "csv", "skip.header.line.count": "1"},
+    },
+}
 
-def get_data_product_metadata_spec_path(version: str = "") -> str:
-    # if version is empty we'll get the latest version
-    if version == "":
-        file_paths = get_filepaths_from_s3_folder(
-            DataProductConfig.metadata_spec_prefix().uri
-        )
+
+def get_data_product_specification_path(
+    spec_type: JsonSchemaName, version: None | str = None
+) -> str:
+    """gets path for given version or latest version of metadata and schema json schema"""
+
+    # if version is empty we get the latest version
+    if version is None:
+        file_paths = get_filepaths_from_s3_folder(specification_prefix(spec_type).uri)
         versions = list(
```

Comment:
Maybe this latter part of the function could be extracted out and reused as `sort_versions` or something.